src/fmtstrformatter.cpp

```diff
@@ -7,43 +7,24 @@
 #include "logger.h"
 #include "ruststring.h"
 
-extern "C" {
-	void* rs_fmtstrformatter_new();
-
-	void rs_fmtstrformatter_free(void* fmt);
-
-	void rs_fmtstrformatter_register_fmt(
-		void* fmt,
-		char key,
-		const char* value);
-
-	char* rs_fmtstrformatter_do_format(
-		void* fmt,
-		const char* format,
-		std::uint32_t width);
-}
-
 namespace newsboat {
 
 FmtStrFormatter::FmtStrFormatter()
+	: rs_object(fmtstrformatter::bridged::create())
 {
-	rs_fmt = rs_fmtstrformatter_new();
-}
-
-FmtStrFormatter::~FmtStrFormatter()
-{
-	rs_fmtstrformatter_free(rs_fmt);
 }
 
 void FmtStrFormatter::register_fmt(char f, const std::string& value)
 {
-	rs_fmtstrformatter_register_fmt(rs_fmt, f, value.c_str());
+	std::string key(1, f);
```

Comment:
Can't the key be passed over as `u8`? If we use a one-element string, we move the knowledge from compile-time to runtime, and then have to resort to tricks like `key.chars().next()`.

Reply:
That is definitely possible, and more in line with the original, non-CXX, FFI binding. Done: https://github.com/newsboat/newsboat/pull/1676/commits/bc6693b96a6312ab8f852a61c9f3b28429c40aef